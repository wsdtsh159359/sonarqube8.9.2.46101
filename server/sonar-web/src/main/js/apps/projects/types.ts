/*
 * SonarQube
 * Copyright (C) 2009-2021 SonarSource SA
 * mailto:info AT sonarsource DOT com
 *
 * This program is free software; you can redistribute it and/or
 * modify it under the terms of the GNU Lesser General Public
 * License as published by the Free Software Foundation; either
 * version 3 of the License, or (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
 * Lesser General Public License for more details.
 *
 * You should have received a copy of the GNU Lesser General Public License
 * along with this program; if not, write to the Free Software Foundation,
 * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
 */
import { ComponentQualifier } from '../../types/component';

export interface Project {
  analysisDate?: string;
  isFavorite?: boolean;
  key: string;
  leakPeriodDate?: string;
  measures: T.Dict<string>;
  name: string;
  projects?: number;
  qualifier: ComponentQualifier;
  tags: string[];
  visibility: T.Visibility;
  needIssueSync?: boolean;
}

export interface Facet {
  [value: string]: number;
}

export interface Facets {
  [property: string]: Facet;
}
