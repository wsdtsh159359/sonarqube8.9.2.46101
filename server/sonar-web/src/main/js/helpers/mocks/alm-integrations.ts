/*
 * SonarQube
 * Copyright (C) 2009-2021 SonarSource SA
 * mailto:info AT sonarsource DOT com
 *
 * This program is free software; you can redistribute it and/or
 * modify it under the terms of the GNU Lesser General Public
 * License as published by the Free Software Foundation; either
 * version 3 of the License, or (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
 * Lesser General Public License for more details.
 *
 * You should have received a copy of the GNU Lesser General Public License
 * along with this program; if not, write to the Free Software Foundation,
 * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
 */
import {
  AzureProject,
  AzureRepository,
  BitbucketProject,
  BitbucketRepository,
  GithubRepository,
  GitlabProject
} from '../../types/alm-integration';

export function mockAzureProject(overrides: Partial<AzureProject> = {}): AzureProject {
  return {
    name: 'azure-project-1',
    description: 'Azure Project',
    ...overrides
  };
}

export function mockAzureRepository(overrides: Partial<AzureRepository> = {}): AzureRepository {
  return {
    name: 'Azure repo 1',
    projectName: 'Azure Project',
    ...overrides
  };
}

export function mockBitbucketProject(overrides: Partial<BitbucketProject> = {}): BitbucketProject {
  return {
    id: 1,
    key: 'project',
    name: 'Project',
    ...overrides
  };
}

export function mockBitbucketRepository(
  overrides: Partial<BitbucketRepository> = {}
): BitbucketRepository {
  return {
    id: 1,
    slug: 'project__repo',
    name: 'Repo',
    projectKey: 'project',
    ...overrides
  };
}

export function mockGitHubRepository(overrides: Partial<GithubRepository> = {}): GithubRepository {
  return {
    id: 'id1234',
    key: 'key3456',
    name: 'repository 1',
    sqProjectKey: '',
    url: 'owner/repo1',
    ...overrides
  };
}

export function mockGitlabProject(overrides: Partial<GitlabProject> = {}): GitlabProject {
  return {
    id: 'id1234',
    name: 'Awesome Project !',
    slug: 'awesome-project-exclamation',
    pathName: 'Company / Best Projects',
    pathSlug: 'company/best-projects',
    sqProjectKey: '',
    url: 'https://gitlab.company.com/best-projects/awesome-project-exclamation',
    ...overrides
  };
}
