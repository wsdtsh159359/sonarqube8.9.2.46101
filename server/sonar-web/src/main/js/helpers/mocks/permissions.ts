/*
 * SonarQube
 * Copyright (C) 2009-2021 SonarSource SA
 * mailto:info AT sonarsource DOT com
 *
 * This program is free software; you can redistribute it and/or
 * modify it under the terms of the GNU Lesser General Public
 * License as published by the Free Software Foundation; either
 * version 3 of the License, or (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
 * Lesser General Public License for more details.
 *
 * You should have received a copy of the GNU Lesser General Public License
 * along with this program; if not, write to the Free Software Foundation,
 * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
 */
import { mockUser } from '../testMocks';

export function mockPermissionGroup(overrides: Partial<T.PermissionGroup> = {}): T.PermissionGroup {
  return {
    name: 'sonar-admins',
    permissions: ['provisioning'],
    ...overrides
  };
}

export function mockPermissionUser(overrides: Partial<T.PermissionUser> = {}): T.PermissionUser {
  return {
    ...mockUser(),
    active: true,
    name: 'johndoe',
    permissions: ['provisioning'],
    ...overrides
  };
}
