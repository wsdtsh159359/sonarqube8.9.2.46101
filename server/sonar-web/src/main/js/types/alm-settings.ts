/*
 * SonarQube
 * Copyright (C) 2009-2021 SonarSource SA
 * mailto:info AT sonarsource DOT com
 *
 * This program is free software; you can redistribute it and/or
 * modify it under the terms of the GNU Lesser General Public
 * License as published by the Free Software Foundation; either
 * version 3 of the License, or (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
 * Lesser General Public License for more details.
 *
 * You should have received a copy of the GNU Lesser General Public License
 * along with this program; if not, write to the Free Software Foundation,
 * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
 */
export const enum AlmKeys {
  Azure = 'azure',
  BitbucketServer = 'bitbucket',
  BitbucketCloud = 'bitbucketcloud',
  GitHub = 'github',
  GitLab = 'gitlab'
}

export interface AlmBindingDefinition {
  key: string;
  url?: string;
}

export interface AzureBindingDefinition extends AlmBindingDefinition {
  personalAccessToken: string;
  url?: string;
}

export interface BitbucketBindingDefinition extends AlmBindingDefinition {
  personalAccessToken: string;
  url: string;
}

export interface BitbucketCloudBindingDefinition extends AlmBindingDefinition {
  clientId: string;
  clientSecret: string;
  workspace: string;
}

export interface GithubBindingDefinition extends AlmBindingDefinition {
  appId: string;
  clientId: string;
  clientSecret: string;
  privateKey: string;
  url: string;
}

export interface GitlabBindingDefinition extends AlmBindingDefinition {
  personalAccessToken: string;
  url?: string;
}

export interface ProjectAlmBindingResponse {
  alm: AlmKeys;
  key: string;
  repository?: string;
  slug?: string;
  summaryCommentEnabled?: boolean;
  monorepo: boolean;
}

export interface ProjectAzureBindingResponse extends ProjectAlmBindingResponse {
  alm: AlmKeys.Azure;
  repository: string;
  slug: string;
  url: string;
}

export interface ProjectBitbucketBindingResponse extends ProjectAlmBindingResponse {
  alm: AlmKeys.BitbucketServer;
  repository: string;
  slug: string;
}

export interface ProjectGitHubBindingResponse extends ProjectAlmBindingResponse {
  alm: AlmKeys.GitHub;
  repository: string;
}

export interface ProjectGitLabBindingResponse extends ProjectAlmBindingResponse {
  alm: AlmKeys.GitLab;
  repository: string;
  url: string;
}

export interface ProjectAlmBindingParams {
  almSetting: string;
  project: string;
  monorepo: boolean;
}

export interface AzureProjectAlmBindingParams extends ProjectAlmBindingParams {
  projectName: string;
  repositoryName: string;
}

export interface BitbucketProjectAlmBindingParams extends ProjectAlmBindingParams {
  repository: string;
  slug: string;
}

export interface BitbucketCloudProjectAlmBindingParams extends ProjectAlmBindingParams {
  repository: string;
}

export interface GithubProjectAlmBindingParams extends ProjectAlmBindingParams {
  repository: string;
  summaryCommentEnabled: boolean;
}

export interface GitlabProjectAlmBindingParams extends ProjectAlmBindingParams {
  repository?: string;
}

export interface AlmSettingsInstance {
  alm: AlmKeys;
  key: string;
  url?: string;
}

export interface AlmSettingsBindingDefinitions {
  [AlmKeys.Azure]: AzureBindingDefinition[];
  [AlmKeys.BitbucketServer]: BitbucketBindingDefinition[];
  [AlmKeys.BitbucketCloud]: BitbucketCloudBindingDefinition[];
  [AlmKeys.GitHub]: GithubBindingDefinition[];
  [AlmKeys.GitLab]: GitlabBindingDefinition[];
}

export interface AlmSettingsBindingStatus {
  alertSuccess: boolean;
  failureMessage: string;
  type: AlmSettingsBindingStatusType;
}

export enum AlmSettingsBindingStatusType {
  Validating,
  Success,
  Failure,
  Warning
}

export function isProjectBitbucketBindingResponse(
  binding: ProjectAlmBindingResponse
): binding is ProjectBitbucketBindingResponse {
  return binding.alm === AlmKeys.BitbucketServer;
}

export function isProjectGitHubBindingResponse(
  binding: ProjectAlmBindingResponse
): binding is ProjectGitHubBindingResponse {
  return binding.alm === AlmKeys.GitHub;
}

export function isProjectGitLabBindingResponse(
  binding: ProjectAlmBindingResponse
): binding is ProjectGitLabBindingResponse {
  return binding.alm === AlmKeys.GitLab;
}

export function isProjectAzureBindingResponse(
  binding: ProjectAlmBindingResponse
): binding is ProjectAzureBindingResponse {
  return binding.alm === AlmKeys.Azure;
}

export function isBitbucketBindingDefinition(
  binding?: AlmBindingDefinition & { url?: string }
): binding is BitbucketBindingDefinition {
  return binding !== undefined && binding.url !== undefined;
}

export function isBitbucketCloudBindingDefinition(
  binding?: AlmBindingDefinition & { clientId?: string; workspace?: string }
): binding is BitbucketCloudBindingDefinition {
  return binding !== undefined && binding.clientId !== undefined && binding.workspace !== undefined;
}

export function isGithubBindingDefinition(
  binding?: AlmBindingDefinition & { appId?: string; url?: string }
): binding is GithubBindingDefinition {
  return binding !== undefined && binding.appId !== undefined && binding.url !== undefined;
}

export function isGitLabBindingDefinition(
  binding?: AlmBindingDefinition | GithubBindingDefinition | BitbucketCloudBindingDefinition
): binding is GitlabBindingDefinition {
  // There's too much overlap with the others. We must not only validate that certain fields are
  // present, we must also validate that others are NOT present. And even so, we cannot be 100%
  // sure, as right now, Azure, Bitbucket Server, and GitLab have the same signature.
  return (
    binding !== undefined &&
    (binding as GithubBindingDefinition).appId === undefined &&
    (binding as BitbucketCloudBindingDefinition).workspace === undefined
  );
}
