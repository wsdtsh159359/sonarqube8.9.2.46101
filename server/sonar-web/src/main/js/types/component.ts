/*
 * SonarQube
 * Copyright (C) 2009-2021 SonarSource SA
 * mailto:info AT sonarsource DOT com
 *
 * This program is free software; you can redistribute it and/or
 * modify it under the terms of the GNU Lesser General Public
 * License as published by the Free Software Foundation; either
 * version 3 of the License, or (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
 * Lesser General Public License for more details.
 *
 * You should have received a copy of the GNU Lesser General Public License
 * along with this program; if not, write to the Free Software Foundation,
 * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
 */
export enum Visibility {
  Public = 'public',
  Private = 'private'
}

export enum ComponentQualifier {
  Application = 'APP',
  Directory = 'DIR',
  Developper = 'DEV',
  File = 'FIL',
  Portfolio = 'VW',
  Project = 'TRK',
  SubPortfolio = 'SVW',
  SubProject = 'BRC',
  TestFile = 'UTS'
}

export enum ProjectKeyValidationResult {
  Valid = 'valid',
  Empty = 'empty',
  TooLong = 'too_long',
  InvalidChar = 'invalid_char',
  OnlyDigits = 'only_digits'
}

export interface TreeComponent extends T.LightComponent {
  id?: string;
  name: string;
  path?: string;
  refId?: string;
  refKey?: string;
  tags?: string[];
  visibility: T.Visibility;
}

export interface TreeComponentWithPath extends TreeComponent {
  path: string;
}

export function isPortfolioLike(
  componentQualifier?: string | ComponentQualifier
): componentQualifier is ComponentQualifier.Portfolio | ComponentQualifier.SubPortfolio {
  return Boolean(
    componentQualifier &&
      [
        ComponentQualifier.Portfolio.toString(),
        ComponentQualifier.SubPortfolio.toString()
      ].includes(componentQualifier)
  );
}

export function isApplication(
  componentQualifier?: string | ComponentQualifier
): componentQualifier is ComponentQualifier.Application {
  return componentQualifier === ComponentQualifier.Application;
}
