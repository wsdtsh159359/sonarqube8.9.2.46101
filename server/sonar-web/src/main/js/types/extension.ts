/*
 * SonarQube
 * Copyright (C) 2009-2021 SonarSource SA
 * mailto:info AT sonarsource DOT com
 *
 * This program is free software; you can redistribute it and/or
 * modify it under the terms of the GNU Lesser General Public
 * License as published by the Free Software Foundation; either
 * version 3 of the License, or (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
 * Lesser General Public License for more details.
 *
 * You should have received a copy of the GNU Lesser General Public License
 * along with this program; if not, write to the Free Software Foundation,
 * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
 */
import { InjectedIntl } from 'react-intl';
import { Store as ReduxStore } from 'redux';
import { Theme } from 'sonar-ui-common/components/theme';
import { Location, Router } from '../components/hoc/withRouter';
import { Store } from '../store/rootReducer';
import { L10nBundle } from './l10n';

export interface ExtensionStartMethod {
  (params: ExtensionStartMethodParameter | string): ExtensionStartMethodReturnType;
}

export interface ExtensionStartMethodParameter {
  store: ReduxStore<Store, any>;
  el: HTMLElement | undefined | null;
  currentUser: T.CurrentUser;
  intl: InjectedIntl;
  location: Location;
  router: Router;
  theme: Theme;
  baseUrl: string;
  l10nBundle: L10nBundle;
}

export type ExtensionStartMethodReturnType = React.ReactNode | Function | void | undefined | null;
