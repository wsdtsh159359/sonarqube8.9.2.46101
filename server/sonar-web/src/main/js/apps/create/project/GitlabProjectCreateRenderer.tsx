/*
 * SonarQube
 * Copyright (C) 2009-2021 SonarSource SA
 * mailto:info AT sonarsource DOT com
 *
 * This program is free software; you can redistribute it and/or
 * modify it under the terms of the GNU Lesser General Public
 * License as published by the Free Software Foundation; either
 * version 3 of the License, or (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
 * Lesser General Public License for more details.
 *
 * You should have received a copy of the GNU Lesser General Public License
 * along with this program; if not, write to the Free Software Foundation,
 * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
 */
import * as React from 'react';
import { translate } from 'sonar-ui-common/helpers/l10n';
import { getBaseUrl } from 'sonar-ui-common/helpers/urls';
import { GitlabProject } from '../../../types/alm-integration';
import { AlmKeys, AlmSettingsInstance } from '../../../types/alm-settings';
import CreateProjectPageHeader from './CreateProjectPageHeader';
import GitlabProjectSelectionForm from './GitlabProjectSelectionForm';
import PersonalAccessTokenForm from './PersonalAccessTokenForm';
import WrongBindingCountAlert from './WrongBindingCountAlert';

export interface GitlabProjectCreateRendererProps {
  canAdmin?: boolean;
  importingGitlabProjectId?: string;
  loading: boolean;
  loadingMore: boolean;
  onImport: (gitlabProjectId: string) => void;
  onLoadMore: () => void;
  onPersonalAccessTokenCreate: (pat: string) => void;
  onSearch: (searchQuery: string) => void;
  projects?: GitlabProject[];
  projectsPaging: T.Paging;
  searching: boolean;
  searchQuery: string;
  settings?: AlmSettingsInstance;
  showPersonalAccessTokenForm?: boolean;
  submittingToken?: boolean;
  tokenValidationErrorMessage?: string;
}

export default function GitlabProjectCreateRenderer(props: GitlabProjectCreateRendererProps) {
  const {
    canAdmin,
    importingGitlabProjectId,
    loading,
    loadingMore,
    projects,
    projectsPaging,
    searching,
    searchQuery,
    settings,
    showPersonalAccessTokenForm,
    submittingToken,
    tokenValidationErrorMessage
  } = props;

  return (
    <>
      <CreateProjectPageHeader
        title={
          <span className="text-middle">
            <img
              alt="" // Should be ignored by screen readers
              className="spacer-right"
              height="24"
              src={`${getBaseUrl()}/images/alm/gitlab.svg`}
            />
            {translate('onboarding.create_project.gitlab.title')}
          </span>
        }
      />

      {loading && <i className="spinner" />}

      {!loading && !settings && (
        <WrongBindingCountAlert alm={AlmKeys.GitLab} canAdmin={!!canAdmin} />
      )}

      {!loading &&
        settings &&
        (showPersonalAccessTokenForm ? (
          <PersonalAccessTokenForm
            almSetting={settings}
            onPersonalAccessTokenCreate={props.onPersonalAccessTokenCreate}
            submitting={submittingToken}
            validationFailed={Boolean(tokenValidationErrorMessage)}
            validationErrorMessage={tokenValidationErrorMessage}
          />
        ) : (
          <GitlabProjectSelectionForm
            importingGitlabProjectId={importingGitlabProjectId}
            loadingMore={loadingMore}
            onImport={props.onImport}
            onLoadMore={props.onLoadMore}
            onSearch={props.onSearch}
            projects={projects}
            projectsPaging={projectsPaging}
            searching={searching}
            searchQuery={searchQuery}
          />
        ))}
    </>
  );
}
