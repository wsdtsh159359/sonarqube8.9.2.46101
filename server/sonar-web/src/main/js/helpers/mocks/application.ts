/*
 * SonarQube
 * Copyright (C) 2009-2021 SonarSource SA
 * mailto:info AT sonarsource DOT com
 *
 * This program is free software; you can redistribute it and/or
 * modify it under the terms of the GNU Lesser General Public
 * License as published by the Free Software Foundation; either
 * version 3 of the License, or (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
 * Lesser General Public License for more details.
 *
 * You should have received a copy of the GNU Lesser General Public License
 * along with this program; if not, write to the Free Software Foundation,
 * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
 */
import { Application, ApplicationPeriod, ApplicationProject } from '../../types/application';
import { Visibility } from '../../types/component';
import { mockBranch } from './branch-like';

export function mockApplication(overrides: Partial<Application> = {}): Application {
  return {
    branches: [mockBranch()],
    key: 'foo',
    name: 'Foo',
    projects: [mockApplicationProject()],
    visibility: Visibility.Private,
    ...overrides
  };
}

export function mockApplicationPeriod(
  overrides: Partial<ApplicationPeriod> = {}
): ApplicationPeriod {
  return {
    date: '2017-10-01',
    project: 'foo',
    projectName: 'Foo',
    ...overrides
  };
}

export function mockApplicationProject(
  overrides: Partial<ApplicationProject> = {}
): ApplicationProject {
  return {
    branch: 'master',
    isMain: true,
    key: 'bar',
    name: 'Bar',
    ...overrides
  };
}
