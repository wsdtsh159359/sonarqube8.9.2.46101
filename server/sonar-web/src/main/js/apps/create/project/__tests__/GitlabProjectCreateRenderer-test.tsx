/*
 * SonarQube
 * Copyright (C) 2009-2021 SonarSource SA
 * mailto:info AT sonarsource DOT com
 *
 * This program is free software; you can redistribute it and/or
 * modify it under the terms of the GNU Lesser General Public
 * License as published by the Free Software Foundation; either
 * version 3 of the License, or (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
 * Lesser General Public License for more details.
 *
 * You should have received a copy of the GNU Lesser General Public License
 * along with this program; if not, write to the Free Software Foundation,
 * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
 */
import { shallow } from 'enzyme';
import * as React from 'react';
import { mockAlmSettingsInstance } from '../../../../helpers/mocks/alm-settings';
import { AlmKeys } from '../../../../types/alm-settings';
import GitlabProjectCreateRenderer, {
  GitlabProjectCreateRendererProps
} from '../GitlabProjectCreateRenderer';

it('should render correctly', () => {
  expect(shallowRender({ loading: true })).toMatchSnapshot('loading');
  expect(shallowRender({ settings: undefined })).toMatchSnapshot('invalid settings');
  expect(shallowRender({ canAdmin: true, settings: undefined })).toMatchSnapshot(
    'invalid settings, admin user'
  );
  expect(shallowRender()).toMatchSnapshot('pat form');
  expect(shallowRender({ showPersonalAccessTokenForm: false })).toMatchSnapshot(
    'project selection form'
  );
  expect(shallowRender({ tokenValidationErrorMessage: 'error' })).toMatchSnapshot(
    'pat validation error'
  );
});

function shallowRender(props: Partial<GitlabProjectCreateRendererProps> = {}) {
  return shallow<GitlabProjectCreateRendererProps>(
    <GitlabProjectCreateRenderer
      canAdmin={false}
      loading={false}
      loadingMore={false}
      onImport={jest.fn()}
      onLoadMore={jest.fn()}
      onPersonalAccessTokenCreate={jest.fn()}
      onSearch={jest.fn()}
      projects={undefined}
      projectsPaging={{ pageIndex: 1, pageSize: 30, total: 0 }}
      searching={false}
      searchQuery=""
      showPersonalAccessTokenForm={true}
      submittingToken={false}
      settings={mockAlmSettingsInstance({ alm: AlmKeys.GitLab })}
      {...props}
    />
  );
}
