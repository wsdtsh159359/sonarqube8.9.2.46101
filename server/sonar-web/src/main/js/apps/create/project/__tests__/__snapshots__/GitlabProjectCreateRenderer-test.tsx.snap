// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render correctly: invalid settings 1`] = `
<Fragment>
  <CreateProjectPageHeader
    title={
      <span
        className="text-middle"
      >
        <img
          alt=""
          className="spacer-right"
          height="24"
          src="/images/alm/gitlab.svg"
        />
        onboarding.create_project.gitlab.title
      </span>
    }
  />
  <WrongBindingCountAlert
    alm="gitlab"
    canAdmin={false}
  />
</Fragment>
`;

exports[`should render correctly: invalid settings, admin user 1`] = `
<Fragment>
  <CreateProjectPageHeader
    title={
      <span
        className="text-middle"
      >
        <img
          alt=""
          className="spacer-right"
          height="24"
          src="/images/alm/gitlab.svg"
        />
        onboarding.create_project.gitlab.title
      </span>
    }
  />
  <WrongBindingCountAlert
    alm="gitlab"
    canAdmin={true}
  />
</Fragment>
`;

exports[`should render correctly: loading 1`] = `
<Fragment>
  <CreateProjectPageHeader
    title={
      <span
        className="text-middle"
      >
        <img
          alt=""
          className="spacer-right"
          height="24"
          src="/images/alm/gitlab.svg"
        />
        onboarding.create_project.gitlab.title
      </span>
    }
  />
  <i
    className="spinner"
  />
</Fragment>
`;

exports[`should render correctly: pat form 1`] = `
<Fragment>
  <CreateProjectPageHeader
    title={
      <span
        className="text-middle"
      >
        <img
          alt=""
          className="spacer-right"
          height="24"
          src="/images/alm/gitlab.svg"
        />
        onboarding.create_project.gitlab.title
      </span>
    }
  />
  <PersonalAccessTokenForm
    almSetting={
      Object {
        "alm": "gitlab",
        "key": "key",
      }
    }
    onPersonalAccessTokenCreate={[MockFunction]}
    submitting={false}
    validationFailed={false}
  />
</Fragment>
`;

exports[`should render correctly: pat validation error 1`] = `
<Fragment>
  <CreateProjectPageHeader
    title={
      <span
        className="text-middle"
      >
        <img
          alt=""
          className="spacer-right"
          height="24"
          src="/images/alm/gitlab.svg"
        />
        onboarding.create_project.gitlab.title
      </span>
    }
  />
  <PersonalAccessTokenForm
    almSetting={
      Object {
        "alm": "gitlab",
        "key": "key",
      }
    }
    onPersonalAccessTokenCreate={[MockFunction]}
    submitting={false}
    validationErrorMessage="error"
    validationFailed={true}
  />
</Fragment>
`;

exports[`should render correctly: project selection form 1`] = `
<Fragment>
  <CreateProjectPageHeader
    title={
      <span
        className="text-middle"
      >
        <img
          alt=""
          className="spacer-right"
          height="24"
          src="/images/alm/gitlab.svg"
        />
        onboarding.create_project.gitlab.title
      </span>
    }
  />
  <GitlabProjectSelectionForm
    loadingMore={false}
    onImport={[MockFunction]}
    onLoadMore={[MockFunction]}
    onSearch={[MockFunction]}
    projectsPaging={
      Object {
        "pageIndex": 1,
        "pageSize": 30,
        "total": 0,
      }
    }
    searchQuery=""
    searching={false}
  />
</Fragment>
`;
